### ISSUE TITLE
Introduce `Flex` component

### ISSUE THREAD

**Issue by jaridmargolin** [airbnb/react-sketchapp#342]
Right now the `Page` component acts a flex container, allowing to render `Artboard`'s and `SymbolMaster`'s in either a column or a row.

I'd like however to have unlimited flexibility rendering Pages by composing `Flex` components. By having some sort of arbitrary `Flex` container, I could layout items on a `Page` in complex grids if necessary.

This would require us introducing a new Component that can be utilized to compute Yoga layout, but doesn't actually result in any backed rendering.

@mathieudutour - I know this goes against some of your comments in #291, but I'm not sure how else to achieve this.

**note:** it doesn't need to be called `Flex`.

**Comment by mathieudutour** [comment]
Yes I can see the use case. Maybe we could make it super clear and have it as a component exposed on the `Page` component like `Page.Container` or something
